bot/seasons/evergreen/fun.py

```diff
@@ -27,6 +40,48 @@ def __init__(self, bot):
             output += getattr(Emojis, terning, '')
         await ctx.send(output)
 
+    @commands.group(name="uwu", aliases=("uwuwize", "uwuify",), invoke_without_command=True)
+    async def uwu_group(self, ctx: Context, *, text: str) -> None:
+        """Commands for making text uwu."""
+        await ctx.invoke(self.convert_uwu, text=text)
+
+    @uwu_group.command(name="convert")
+    async def convert_uwu(self, ctx: Context, *, text: str) -> None:
+        """Converts a given `text` into it's uwu equivalent."""
+        text = await Fun.text_to_message(ctx, text)
+        converted = utils.replace_many(text, UWU_WORDS, ignore_case=True, match_case=True)
+        await ctx.send(f">>> {converted}")
+
+    @commands.group(name="randomcase", aliases=("rcase", "randomcaps", "rcaps",), invoke_without_command=True)
+    async def randomcase_group(self, ctx: Context, *, text: str) -> None:
+        """Commands for returning text in randomcase."""
+        await ctx.invoke(self.convert_randomcase, text=text)
+
+    @randomcase_group.command(name="convert")
+    async def convert_randomcase(self, ctx: Context, *, text: str) -> None:
+        """Randomly converts the casing of a given `text`."""
+        text = await Fun.text_to_message(ctx, text)
+        converted = (
+            char.upper() if round(random.random()) else char.lower() for char in text
+        )
+        await ctx.send(f">>> {''.join(converted)}")
+
+    @staticmethod
+    async def text_to_message(ctx: Context, text: str) -> str:
+        """
+        Attempts to convert a given `text` to a discord Message, then return the contents.
+
+        Returns `text` if the conversion fails.
+        """
+        try:
+            message = await MessageConverter().convert(ctx, text)
```

Comment:
Why is this conversion needed? How would the contents after the conversion differ from the input and why do we care?

Reply:
What was the answer to this question?

Reply:
Never mind, reading the docstring would help 💯

Reply:
Whoops, answered it elsewhere. It's so the commands work with a valid Discord message ID or link as an argument, if the conversion suceeds the text from the linked message gets converted.